File: 0905-sort-array-by-parity/0905-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        n = len(nums)
        r = n - 1
        i = 0

        while i <= r:
            if nums[i] % 2 == 0:
                i += 1
            else:
                nums[i], nums[r] = nums[r], nums[i]
                r -= 1
        
        return nums

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
    let lp = 0;
    let hp = nums.length-1;
    
    while (lp<hp) {
        let lEven = nums[lp] % 2 === 0;
        let hOdd = nums[hp] % 2 !== 0;
        
        if(lEven && hOdd){
            lp++;
            hp--
        } else if(!lEven && !hOdd){
            let temp = nums[lp];
            nums[lp] = nums[hp];
            nums[hp] = temp
        } else if(lEven && !hOdd){
            lp++;
        } else if(!lEven && hOdd){
            hp--;
        }
    }
    return nums
};